alacritty/src/input.rs

```diff
@@ -917,16 +941,43 @@ impl<T: EventListener, A: ActionContext<T>> Processor<T, A> {
         }
     }
 
+    /// Attempt to find a binding and execute its action.
+    ///
+    /// The provided mode, mods, and key must match what is allowed by a binding
+    /// for its action to be executed.
+    fn process_mouse_bindings(&mut self, button: MouseButton) {
+        let mode = BindingMode::new(self.ctx.terminal().mode(), self.ctx.search_active());
+        let mouse_mode = self.ctx.mouse_mode();
+        let mods = self.ctx.modifiers().state();
+
+        for i in 0..self.ctx.config().mouse_bindings().len() {
+            let mut binding = self.ctx.config().mouse_bindings()[i].clone();
+
+            // Require shift for all modifiers when mouse mode is active.
+            if mouse_mode {
+                binding.mods |= ModifiersState::SHIFT;
+            }
+
+            if binding.is_triggered_by(mode, mods, &button) {
+                binding.action.execute(&mut self.ctx);
+            }
+        }
+    }
+
     /// Process key input.
-    pub fn key_input(&mut self, input: KeyboardInput) {
+    pub fn key_input(&mut self, key: KeyEvent) {
         // IME input will be applied on commit and shouldn't trigger key bindings.
-        if self.ctx.display().ime.preedit().is_some() {
+        if key.state == ElementState::Released || self.ctx.display().ime.preedit().is_some() {
             return;
         }
 
+        let text = key.text_with_all_modifiers().unwrap_or_default();
+
         // All key bindings are disabled while a hint is being selected.
         if self.ctx.display().hint_state.active() {
-            *self.ctx.suppress_chars() = false;
+            for ch in text.chars() {
+                self.ctx.hint_input(ch);
+            }
```

Comment:
```suggestion
            for c in text.chars() {
                self.ctx.hint_input(c);
            }
```

Far more common abbreviation I think.